Last-Update: 2012-09-08
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: not-needed
Description: backported patch (r31861) to fix FTBFS with --enable-collab-backend-fake

--- a/plugins/collab/backends/fake/xp/FakeAccountHandler.cpp
+++ b/plugins/collab/backends/fake/xp/FakeAccountHandler.cpp
@@ -295,7 +295,7 @@
 	{
 		UT_DEBUGMSG(("Starting a locally controlled collaboration session: %s\n", sSessionId.utf8_str()));
 		// FIXME: we need to set the proper collab id and master descriptor
-		m_pSession = pManager->startSession(m_pDoc, sSessionId, NULL, "fake://master");
+		m_pSession = pManager->startSession(m_pDoc, sSessionId, this, true, NULL, "fake://master");
 	}
 	else
 	{
@@ -320,8 +320,7 @@
 		}
 		
 		addBuddy(pCollaborator);
-		XAP_Frame* pFrame = XAP_App::getApp()->newFrame(); // FIXME: this is a memory leak
-		m_pSession = new AbiCollab(sSessionId, m_pDoc, sDocUUID /* FIXME: this is the local doc uuid, is that valid?? */, jsrre.m_iRev, pCollaborator, pFrame);
+		m_pSession = new AbiCollab(sSessionId, m_pDoc, sDocUUID /* FIXME: this is the local doc uuid, is that valid?? */, jsrre.m_iRev, pCollaborator, this, true);
 		pManager->joinSession(m_pSession, pCollaborator);
 	}
 
--- a/plugins/collab/backends/fake/xp/FakeAccountHandler.h
+++ b/plugins/collab/backends/fake/xp/FakeAccountHandler.h
@@ -64,6 +64,8 @@
 		{ return false; }
 	virtual void							forceDisconnectBuddy(BuddyPtr pBuddy);
 	virtual bool							recognizeBuddyIdentifier(const std::string& identifier);
+	virtual bool							hasPersistentAccessControl()
+		{ return true; }
 
 	// session management
 	virtual bool							allowsSessionTakeover()
--- a/plugins/collab/core/session/xp/AbiCollabSessionManager.cpp
+++ b/plugins/collab/core/session/xp/AbiCollabSessionManager.cpp
@@ -94,6 +94,9 @@
 #ifdef ABICOLLAB_HANDLER_SIPSIMPLE
 #include <backends/sipsimple/unix/SIPSimpleUnixAccountHandler.h>
 #endif
+#ifdef ABICOLLAB_HANDLER_FAKE
+#include <backends/fake/xp/FakeAccountHandler.h>
+#endif
 
 // event includes
 #include <account/xp/Event.h>
@@ -287,6 +290,10 @@
 #ifdef ABICOLLAB_HANDLER_SIPSIMPLE
 	m_regAccountHandlers[SIPSimpleAccountHandler::getStaticStorageType()] = SIPSimpleAccountHandlerConstructor;
 #endif
+#ifdef ABICOLLAB_HANDLER_FAKE
+	AccountHandler *pFakeHandler = new FakeAccountHandler("FakeAbiCollabBackend", NULL);
+	addAccount(pFakeHandler);
+#endif
 	return true;
 }
 
